common/src/main/java/com/wynntils/services/itemfilter/filters/RangedStatFilters.java

```diff
@@ -91,26 +92,36 @@ protected RangedStatValueStatFilter getRangedStatFilter(int min, int max) {
     }
 
     private abstract static class RangedStatFilterFactory<T> extends StatFilterFactory<T> {
-        private static final Pattern SINGLE_VALUE_PATTERN = Pattern.compile("\\d+");
-        private static final Pattern RANGE_PATTERN = Pattern.compile("\\d+-\\d+");
-        private static final Pattern GREATER_THAN_PATTERN = Pattern.compile(">(=)?\\d+");
-        private static final Pattern LESS_THAN_PATTERN = Pattern.compile("<(=)?\\d+");
+        private static final Pattern SINGLE_VALUE_PATTERN = Pattern.compile("[-+]?\\d+");
+        private static final Pattern RANGE_PATTERN = Pattern.compile("([-+]?\\d+)-([-+]?\\d+)");
+        private static final Pattern GREATER_THAN_PATTERN = Pattern.compile(">(=)?[-+]?\\d+");
+        private static final Pattern LESS_THAN_PATTERN = Pattern.compile("<(=)?[-+]?\\d+");
 
         @Override
         public Optional<T> create(String inputString) {
-            if (SINGLE_VALUE_PATTERN.matcher(inputString).matches()) {
+            Matcher matcher = SINGLE_VALUE_PATTERN.matcher(inputString);
+
+            if (matcher.matches()) {
                 int value = Integer.parseInt(inputString);
                 return Optional.of(getRangedStatFilter(value, value));
-            } else if (RANGE_PATTERN.matcher(inputString).matches()) {
-                String[] split = inputString.split("-");
-                int min = Integer.parseInt(split[0]);
-                int max = Integer.parseInt(split[1]);
+            }
+
+            matcher = RANGE_PATTERN.matcher(inputString);
+            if (matcher.matches()) {
+                int min = Integer.parseInt(matcher.group(1));
+                int max = Integer.parseInt(matcher.group(2));
```

Comment:
Yea this is much nicer, dunno why we used a split before

Reply:
Yeah, my bad.